### PR TITLE
SONARJAVA-3470 Fix FP rule S107 by adding more ignored annotations

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**

### CODE REVIEW

**Comment by margarita-nedzelska-sonarsource** [comment]
In the ticket there is a suggestion to add one more parameter to the rule but actually I have doubts whether it will bring much value

**Comment by margarita-nedzelska-sonarsource** [comment]
After discussing with @m-g-sonar we decided not to add an additional parameter to the rule, because all the cases were about JAX-RS annotations and we covered it here. And The general rule is to keep checks simple.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S107_java.html [hunk 1]
```diff
@@ -14,6 +14,11 @@ <h2>Compliant Solution</h2>
...
 </pre>
 <h2>Exceptions</h2>
-<p>Methods annotated with Spring's <code>@RequestMapping</code> (and related shortcut annotations, like <code>@GetRequest</code>) or
-<code>@JsonCreator</code> may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.</p>
+<p>Methods annotated with :</p>
+<p> * Spring's <code>@RequestMapping</code> (and related shortcut annotations, like <code>@GetRequest</code>)</p>
+<p> * JAX-RS API annotations (like <code>@javax.ws.rs.GET</code>)</p>
+<p> * Bean constructor injection with <code>@org.springframework.beans.factory.annotation.Autowired</code></p>
+<p> * CDI constructor injection with <code>@javax.inject.Inject</code></p>
+<p> * <code>@com.fasterxml.jackson.annotation.JsonCreator</code></p>
```
Comment (quentin-jaquier-sonarsource) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S107_java.html:22]: This should be an HTML list.
It's a ruleapi bug, you have to remove the space before the `*` in the description.

I already did it, you just have to generate the metadata to solve the problem.

**Review by quentin-jaquier-sonarsource — APPROVED** [review]
Also make sure that the [description of the ticket](https://jira.sonarsource.com/browse/SONARJAVA-3470) is up to date with the current list.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S107_java.html (hunk 1 above)
Comment (margarita-nedzelska-sonarsource) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S107_java.html]: Thanks, didn't know about it 👍

**Pull request merged**